app/controllers/declaration/DestinationCountriesController.scala

```diff
@@ -82,23 +84,24 @@ class DestinationCountriesController @Inject()(
     implicit request: JourneyRequest[AnyContent],
     hc: HeaderCarrier
   ): Future[Result] =
-    supplementaryForm
+    Supplementary.form
       .bindFromRequest()
       .fold(
-        (formWithErrors: Form[DestinationCountriesSupplementary]) =>
+        (formWithErrors: Form[DestinationCountries]) =>
           Future.successful(BadRequest(destination_countries_supplementary(formWithErrors))),
-        form =>
-          customsCacheService.cache[DestinationCountriesSupplementary](cacheId, formId, form).map { _ =>
-            Redirect(controllers.declaration.routes.LocationController.displayForm())
-        }
+        formData =>
+          for {
+            _ <- updateCache(journeySessionId, formData)
+            _ <- customsCacheService.cache[DestinationCountries](cacheId, formId, formData)
+          } yield Redirect(controllers.declaration.routes.LocationController.displayForm())
       )
 
   private def handleSubmitStandard()(implicit request: JourneyRequest[AnyContent]): Future[Result] = {
     val actionTypeOpt = request.body.asFormUrlEncoded.map(FormAction.fromUrlEncoded)
 
     val cachedData = customsCacheService
-      .fetchAndGetEntry[DestinationCountriesStandard](cacheId, formId)
-      .map(_.getOrElse(DestinationCountriesStandard.empty))
+      .fetchAndGetEntry[DestinationCountries](cacheId, formId)
```

Comment:
This is changing the behaviour of the old caching which is risky and not the purpose of the caching approach. I'm not a fan of the changes made here in the fact that although the models/forms differ by one field we may in future decide to split the journey out to its own service. By coupling these we make it harder to see the journeys differ.

Reply:
We cant convert this controller to the new cache until this or another approach to fixing this is done @mattclark-zerogravit. It only works currently because the "pulling the data out of the cache" bit wasn't implemented fully for the Standard Journey

Reply:
I'd like to understand better why you think this changes the behaviour @mattclark-zerogravit. All the original tests passed, so as far as i'm aware existing behaviour has been preserved